pg_lakehouse/tests/table_config.rs

```diff
@@ -148,3 +148,28 @@ async fn test_preserve_casing(mut conn: PgConnection, tempdir: TempDir) -> Resul
 
     Ok(())
 }
+
+#[rstest]
+async fn test_preserve_casing_on_table_create(
+    mut conn: PgConnection,
+    tempdir: TempDir,
+) -> Result<()> {
+    let stored_batch = primitive_record_batch()?;
+    let parquet_path = tempdir.path().join("test_arrow_types.parquet");
+    let parquet_file = File::create(&parquet_path)?;
+
+    let mut writer = ArrowWriter::try_new(parquet_file, stored_batch.schema(), None).unwrap();
+    writer.write(&stored_batch)?;
+    writer.close()?;
+
+    primitive_setup_fdw_local_file_listing(parquet_path.as_path().to_str().unwrap(), "MyTable")
+        .execute(&mut conn);
+
+    format!(
```

Comment:
This test is missing two checks:

1. Check that auto schema creation (ie `CREATE FOREIGN TABLE \"MyTable\" () SERVER parquet_server OPTIONS (files '{}', preserve_casing 'true')`) preserves the table case
2. Test that table casing is actually preserved by doing `SELECT * FROM "MyTable"` and ensuring that query executes successfully

Reply:
@rebasedming please take a look again. I have the new test cases implemented :)